x-pack/plugins/endpoint/public/embeddables/resolver/store/data/selectors.ts

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createSelector } from 'reselect';
+import { ResolverState } from '../../types';
+import { sampleData } from './sample';
+import { depthFirstPreorder, levelOrder } from '../../lib/tree_sequencers';
+import { Vector2 } from '../../types';
+import { add as vector2Add } from '../../lib/vector2';
+
+const unit = 100;
+const distanceBetweenNodesInUnits = 1;
+const distanceBetweenNodes = distanceBetweenNodesInUnits * unit;
+
+function dataSelector(state: ResolverState) {
```

Comment:
Check out this comment I made on Paul's PR: https://github.com/elastic/kibana/pull/53906/files#r363367614
I explain my ideas around organizing selectors. Can we do that here? Basic plan:

1. Use a reducer just for this concern, let's call it `dataReducer` for this example. It takes `DataState` and `ResolverAction` and returns a `DataState`. This reducer doesn't know anything about global state.
2. Have 'data' selectors that take `DataState`. They don't know anything about global state. These can be called by the `dataReducer` since the data reducer takes `DataState` and these selectors take `DataState`. These are sorta like model methods for `dataState`
3. Import and re-expose some data selectors from `store/selectors`. The selectors in `store/selectors` take Global state and are the only selectors used by react components. This is because react components receive global state from useSelector, and they should have no knowledge of the internals of the global state. Since the 'data' selectors take 'DataState' and not global state, you will need to create a version of your selectors that takes GlobalState and then calls the underlying data selectors with the right piece of it. I'm using a helper i made called `composeSelectors`. Here it is in my PR: https://github.com/elastic/kibana/pull/53619/files#diff-7dd777f1d03cd310741b16bdfb94291fR28
4. Use only the selectors in `store/selectors` from react components. These selectors can be thought of as 'public' or you could think of them as model methods for GlobalState. 
5. If some business logic isn't really specific to any particular state, but instead operates on a well defined business construct (e.g. Event, ProcessEvent, User) then place that logic in `models/whatever`. These types of functions can be pure (they don't rely on state per say, just an entity that could be found anywhere.) They should be easy to test as well, without state there is no need for redux/actions/etc.

Reply:
Done